src/BinaryParsers/ElfBinary/Dwarf/DwarfCompilationUnit.cs

```diff
@@ -29,10 +31,23 @@ public class DwarfCompilationUnit
         /// <param name="debugData">The debug data stream.</param>
         /// <param name="debugDataDescription">The debug data description stream.</param>
         /// <param name="debugStrings">The debug strings.</param>
+        /// <param name="debugStrings">The debug string offsets.</param>
         /// <param name="addressNormalizer">Normalize address delegate (<see cref="NormalizeAddressDelegate"/>)</param>
-        public DwarfCompilationUnit(IDwarfBinary dwarfBinary, DwarfMemoryReader debugData, DwarfMemoryReader debugDataDescription, DwarfMemoryReader debugStrings, NormalizeAddressDelegate addressNormalizer)
+        public DwarfCompilationUnit(IDwarfBinary dwarfBinary,
+                                    DwarfMemoryReader debugData,
+                                    DwarfMemoryReader debugDataDescription,
+                                    DwarfMemoryReader debugStrings,
+                                    DwarfMemoryReader debugLineStrings,
+                                    IList<int> debugStringOffsets,
```

Comment:
>debugStringOffsets [](http://example.com/codeflow?start=47&length=18)

So this implementation today processes and resolves all debug string offsets. That's because parsing/interpreting this data seems to be different than what the current design can be cleanly extended to include. I'll keep looking at it.

The way this works is that there's a single debug string offsets section. Each element in this section comprises a string location. The attribute values include an index into this locations array. Each compilation unit, however, has an index offset (which marks where in the overall section the compilation unit started contributing strings).

As a convenience, we rip through the section completely to make it easier to resolve things. The remainder of the code works in a sort of forward only capacity (often caching data that its already parsed).